### PR TITLE
Test int->string type overrides with cross-tests

### LINKED ISSUE #2048: Testing: Add an integration test for type overrides
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Add an integration test for the behavior added in https://github.com/pulumi/pulumi-terraform-bridge/pull/1896.

Follow up on https://github.com/pulumi/home/issues/3421

### PR DESCRIPTION
**Pull request by iwahbe**
Stacked PRs:
 * __->__#2502
 * #2503
 * #2501
 * #2500

--- --- ---

### Test int->string type overrides with cross-tests

Fixes #2048

### CODE REVIEW

**Review by VenelinMartinov — COMMENTED** [review]
Don't want to block this on the refactoring work.

If you want to, you can implement this as a "manual" cross-test like so: https://github.com/pulumi/pulumi-terraform-bridge/blob/bf282ab5f0663bf7813ed248a4e9c81c36ee4587/pkg/tests/schema_pulumi_test.go#L1593

File: pkg/tfbridge/tests/provider_test.go [hunk 1]
```diff
@@ -5,21 +5,47 @@ import (
...
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfgen"
 	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
 )
 
+func TestIntToStringOverride(t *testing.T) {
+	t.Parallel()
+
+	const largeInt = math.MaxInt64 / 2
+
+	crosstests.Create(t,
```
Comment (t0yv0) [inline pkg/tfbridge/tests/provider_test.go:33]: GPT conversation.

I am designing a Go library to test that Pulumi and Terraform programs behave identically. I want to design a Golang
method that checks that for a given set of inputs provided by the user, Pulumi and Terraform will create resources with
identical data flowing into the provider code handling resource creation. How can I name this method so the name is
short but meaningful?



Here are a few suggestions for naming the method:

1. =VerifyIdenticalOutputs=
2. =CompareResourceData=
3. =CheckConvergedStates=
4. =ValidateOutputsMatch=
5. =MatchProviderInputs=

Each name emphasizes different aspects, such as verification, comparison, checking, validation, or matching, while aiming to convey the core functionality concisely.

Comment (t0yv0) [reply pkg/tfbridge/tests/provider_test.go:33]: That didn't help. GPT is bad at naming things just as we are. `crosstests.VerifyCreate` possibly.

**Review by t0yv0 — APPROVED** [review]

File: pkg/tfbridge/tests/provider_test.go (hunk 1 above)
Comment (iwahbe) [reply pkg/tfbridge/tests/provider_test.go:33]: Maybe rename `crosstests` to `crosstest`, then it's `crosstest.Create`, which reads well. We can do naming in a follow-up PR. I want to keep the pattern consistent between SDKv2 and PF based cross-tests.

Comment (t0yv0) [reply pkg/tfbridge/tests/provider_test.go:33]: Curious you mention this, why are SDKv2 and PF based cross tests any different? Seems like you can take a somehow derived TF provider (SDKv2 or PF), and some inputs, and cross-check TF vs Pulumi behavior, without caring which one it is.

Comment (t0yv0) [reply pkg/tfbridge/tests/provider_test.go:33]: We probably have some legacy gunk to make it happen though.. but in principle..

Comment (t0yv0) [reply pkg/tfbridge/tests/provider_test.go:33]: NVM, I'm a little tired, perhaps that's exactly what you are doing already

**Pull request merged**